domain/lib/src/model/document/document.dart

```diff
@@ -48,6 +48,8 @@ class Document extends Equatable {
   final bool hasThumbnail;
   final int shared;
   final MediaType mediaType;
+  final String localPath;
```

Comment:
Are your PR target to Flutter 2.0. IMO, it should be `String?`. @huynguyennovem please take a look on it.

Reply:
This does not support null-safety yet. Please don't care here.

Reply:
It's not required to mark it as nullable. It depends on its setter. 
BTW, if `localPath` is String, it does not need to check null here:
```
bool isOfflineMode() => localPath != null && localPath.isNotEmpty;
```

Reply:
> This does not support null-safety yet. Please don't care here.

After null-safety is merged. All features MUST be support null-safety.

Reply:
> It's not required to mark it as nullable. It depends on its setter.
> BTW, if `localPath` is String, it does not need to check null here:
> 
> ```
> bool isOfflineMode() => localPath != null && localPath.isNotEmpty;
> ```

But in some cases, Ex: Document is get from server, this property is not set. So it must be nullable.

Reply:
Document here is not an object retrieved from the server. Another object is retrieved from the server and converted to it.